lib/jake/ref.ex

```diff
@@ -1,7 +1,49 @@
 defmodule Jake.Ref do
   alias Jake.Context
 
+  def expand_ref(%Context{child: %{"$ref" => ref} = spec} = context)
+      when ref == "#" do
+    new_child = Map.drop(spec, ["$ref"]) |> Map.merge(context.root)
+    {%{context | child: new_child}, true}
+  end
+
+  def expand_ref(%Context{child: %{"$ref" => ref} = spec} = context) when is_binary(ref) do
+    uri = URI.decode(ref)
+
```

Comment:
could we just use URI.parse here? it seems to be able to parse non http uri also. also, I think `abc.json#foo.bar` means we need to load abc.json file and look for foo.bar inside

Reply:
> could we just use URI.parse here? 

yes, URI.parse will simplify many things here.

If we have something like `abc.json#/foo/bar`, then it means we've to look into `abc.json` for `/foo/bar`. I've also verified it from specs. My question is how should we locate this `abc.json` locally? Should we setup some path initially, where we should look for this file.

Reply:
we could make it configurable, and probably could consider the current working directory as the default root directory

Reply:
ok that's fine.